packages/shared/charts/src/components/Chart.tsx

```diff
@@ -26,9 +30,16 @@ export const Chart = ({
     customConfig,
     customLayout
 }: ChartProps): ReactElement => {
-    const forceReactPlotlyUpdate = useCallback(() => {
-        window.dispatchEvent(new Event("resize"));
-    }, []);
+    const hasForceUpdatedReactPlotly = useRef(false);
+
+    useEffect(() => {
+        const dataIsLoaded = data.length > 0;
+        // The lib doesn't autosize the chart properly in the beginning, so we manually trigger a refresh once when everything is ready.
+        if (!hasForceUpdatedReactPlotly.current && dataIsLoaded) {
+            forceReactPlotlyUpdate();
+            hasForceUpdatedReactPlotly.current = true;
+        }
+    }, [data]);
```

Comment:
Does it means every data update it will resize?

Reply:
No, we only want to force update once when the data turns from "non loaded" to "loaded". That's why we need the `hasForceUpdatedReactPlotly` ref, which is initialized to false. On every `data` change we **check** whether it's necessary to force update. If it hasn't been force updated before (which means the data wasn't ready before), it will update. If not (which means the current change in data is something that happened on the fly), it won't.

Reply:
👍🏻